src/server/req_modify.c

```diff
@@ -862,6 +859,17 @@ req_modifyReservation(struct batch_request *preq)
 					req_reject(PBSE_STDG_RESV_OCCR_CONFLICT, 0, preq);
 					return;
 				}
+
+				break;
+			case RESV_ATR_duration:
+				if (presv->ri_alter_etime == 0)
+					presv->ri_alter_flags |= RESV_START_TIME_MODIFIED;
+					presv->ri_alter_etime = presv->ri_wattr[RESV_ATR_end].at_val.at_long;
+					if (presv->ri_alter_stime == 0)
+						presv->ri_alter_stime = presv->ri_wattr[RESV_ATR_start].at_val.at_long;
```

Comment:
This should be like the other attributes, where you do something like:
```C
temp = strtol(psatl->al_value, &end, 10);
presv->ri_alter_stime = presv->ri_wattr[RESV_ATR_start].at_val.at_long;
presv->ri_alter_flags |= RESV_DURATION_MODIFIED;
send_to_scheduler = RESV_DURATION_MODIFIED;
```
Do you need to check if it's a standing reservation and will collide with the next occurrance?

Reply:
added a check for this. Also added a test